checksum/checksum.go

```diff
@@ -123,6 +119,7 @@ func Calculate(c *Checksum, t Type, v []byte) {
 	case TZ:
 		c.SetTillichZemor(tz.Sum(v))
 	default:
+		panic(fmt.Errorf("unsupported checksum type %v", t))
```

Comment:
i am not sure it is good. we [have discussed panics](https://github.com/nspcc-dev/neofs-sdk-go/issues/390) in SDK and i thought we do not use them anymore. if a wrong scheme _must be_ (really?) noticed, i would prefer an error

Reply:
panics are butter than hidden bugs. We discussed this for functions which return an error. Here all we can do is panic. I'll add new function (will try to find an appropriate name) and mark this as deprecated

Reply:
ok, i reopened this issue with some my thoughts: https://github.com/nspcc-dev/neofs-sdk-go/issues/390#issuecomment-2218503588